src/bridge/settings/openedx/mfe/slot_config/Footer.jsx

```diff
@@ -135,6 +136,117 @@ const ForceLoginRedirect = () => {
   return null;
 };
 
+const AutoSelectLanguage = () => {
+  const config = getConfig();
+  const location = useLocation();
+  const { authenticatedUser } = useContext(AppContext);
+
+  const lmsBaseURL = config.LMS_BASE_URL;
+  const studioBaseURL = config.STUDIO_BASE_URL;
+  const username = authenticatedUser?.username;
+  const DEVELOPMENT_ENVIRONMENT = "development";
+  const AUTHORING_APP_ID = "authoring";
+  const ENGLISH_LANG_CODE = "en";
+  const LANGUAGE_PREFERENCE_COOKIE_NAME =
+    config.LANGUAGE_PREFERENCE_COOKIE_NAME ||
+    process.env.LANGUAGE_PREFERENCE_COOKIE_NAME ||
+    `${process.env.ENVIRONMENT}-openedx-language-preference`;
+  const courseKeyRegex = /course-v1:[^/]+/;
+  const reloadCookieName = "authoringLangCookieReloaded";
+
+  // Helper to escape regex metacharacters in cookie name
+  const escapeRegExp = (string) => string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+  const getCookie = (name) => {
+    const safeName = escapeRegExp(name);
+    const match = document.cookie.match(new RegExp(`(^| )${safeName}=([^;]+)`));
+    return match ? decodeURIComponent(match[2]) : null;
+  };
+  const setCookie = (name, value, days = 1) => {
+    const expires = new Date(Date.now() + days * 864e5).toUTCString();
+    document.cookie = `${name}=${encodeURIComponent(value)}; expires=${expires}; path=/`;
+  };
+  const removeCookie = (name) => {
+    document.cookie = `${name}=; expires=Thu, 01 Jan 1970 00:00:00 UTC; path=/;`;
+  };
+
+  async function setLanguage(baseURL, language) {
+    try {
+      const response = await getAuthenticatedHttpClient().patch(
+        `${baseURL}/api/user/v1/preferences/${username}`,
+        { "pref-lang": language },
+        { headers: { "Content-Type": "application/merge-patch+json" } }
+      );
+      return response.status === 204;
+    } catch (error) {
+      if (process.env.NODE_ENV === DEVELOPMENT_ENVIRONMENT) {
+        console.error("Language update failed:", error);
+      }
+      return false;
+    }
+  }
+
+  async function fetchCourseLanguage(courseKey) {
+    const url = `${lmsBaseURL}/course-translations/api/course-language/${courseKey}`;
```

Comment:
For posterity, can we add a code comment and mention that this URL comes from our own plugin? We can give the URL of our plugin too or at least the name.